sematic/api/endpoints/payloads.py

```diff
@@ -0,0 +1,60 @@
+"""Augment or modify the returned json for ORM models for returns from API calls"""
+# Standard Library
+import logging
+from typing import Any, Dict, List
+
+# Sematic
+from sematic.db.models.resolution import Resolution
+from sematic.db.models.run import Run
+from sematic.db.queries import get_users
+
+logger = logging.getLogger(__name__)
+
+
+def get_runs_payload(runs: List[Run]) -> List[Dict[str, Any]]:
+    """
+    Build the standard payload for a list of runs.
+    """
+    runs_payload = [run.to_json_encodable() for run in runs]
+
+    _set_user_payloads(runs_payload)
+
+    return runs_payload
+
+
+def get_run_payload(run: Run) -> Dict[str, Any]:
+    """
+    Build the standard payload for a single run.
+    """
+    run_payload = run.to_json_encodable()
+
+    _set_user_payloads([run_payload])
+
+    return run_payload
+
+
+def get_resolution_payload(resolution: Resolution) -> Dict[str, Any]:
+    """
+    Build the standard payload for a single resolution.
+    """
+    resolution_payload = resolution.to_json_encodable()
+
+    _set_user_payloads([resolution_payload])
+
+    return resolution_payload
+
+
+def _set_user_payloads(items: List[Dict[str, Any]]):
+    try:
```

Comment:
it might be helpful to mention that this is a typical job of GraphQL or any middleware Backend For the Frontend. It is very natural to handle this kind of job efficiently and concisely there. 

Just wanna mention it, please don't revert the change or attempt to change to GQL. But it might be helpful to keep this in mind when the app grows larger and we have to stratify.

Reply:
Yes, it is true, GQL does that.
I have also heard that GQL can become a hell-hole of low performance and inefficiency. So maybe better to DIY?

Reply:
If you have a reference on the `low performance and inefficiency`, I am very interested to go over it :)